### PR TITLE
Example with next.js 13.4 app directory written in typescript

### PR DESCRIPTION
**Pull request by trace2798**

### CODE REVIEW

File: 03-Login-app-directory/components/LogoutButton.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"use client";
+import { useRouter } from "next/navigation";
+import { FC, MouseEventHandler } from "react";
+
+interface LogoutButtonProps {}
+
+const LogoutButton: FC<LogoutButtonProps> = ({}) => {
+  const router = useRouter();
+  const handleLogout: MouseEventHandler<HTMLDivElement> = () => {
+    localStorage.clear();
```
Comment (himichaelroberts) [inline 03-Login-app-directory/components/LogoutButton.tsx:10]: In our [Passage.js](https://docs.passage.id/frontend/passage.js/session-reference) library, we now have a session class to help manage the user's session.

**Review by himichaelroberts — COMMENTED** [review]
Thanks so much for contributing a Next.js App Router example for us! 

I left a few comments, but overall great example.

File: 03-Login-app-directory/actions/getCurrentUser.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { cookies } from "next/headers";
+
+export async function getCurrentUser() {
+  const passage = new Passage({
+    appID: process.env.NEXT_PUBLIC_PASSAGE_APP_ID!,
+    apiKey: process.env.NEXT_PUBLIC_PASSAGE_API_KEY!,
+    authStrategy: "HEADER",
+  });
+  // console.log(passage.appID, passage.apiKey)
+  // console.log("after passage")
```
[the remark this answers is not in the document]
Comment (trace2798) [reply 03-Login-app-directory/actions/getCurrentUser.ts:11]: Hey. Do you want me to delete all the commented lines? Also are you "passenger-michael#2898" on discord?

File: 03-Login-app-directory/actions/getCurrentUser.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import Passage from "@passageidentity/passage-node";
+import { cookies } from "next/headers";
+
+export async function getCurrentUser() {
+  const passage = new Passage({
+    appID: process.env.NEXT_PUBLIC_PASSAGE_APP_ID!,
+    apiKey: process.env.NEXT_PUBLIC_PASSAGE_API_KEY!,
```
[the remark this answers is not in the document]
Comment (trace2798) [reply 03-Login-app-directory/actions/getCurrentUser.ts:7]: I will work on this now.

**Comment by trace2798** [comment]
Hey in my latest commit, I changed the server action file to use @passageidentity/passage-js instead of "@passageidentity/passage-node. Now since I am using @passageidentity/passage-js I just need the appID. Let me know if I need to make more changes.

File: 03-Login-app-directory/actions/getCurrentUserInfo.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import { Passage } from "@passageidentity/passage-js";
+
+export interface PassageUserInfo {
+  email: string;
+  created_at: string;
+}
+
+export async function getCurrentUserInfo() {
```
Comment (himichaelroberts) [inline 03-Login-app-directory/actions/getCurrentUserInfo.ts:8]: This is perfect for checking currentUser client side.

**Comment by himichaelroberts** [comment]
The changes look perfect for checking the user auth client-side.

There is still value in providing an example of how one could verify JWTs server-side. A common use case would be for API routes. [Next.js middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) would be perfect for handling this.

File: 03-Login-app-directory/middleware.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import { NextResponse } from "next/server";
+import type { NextRequest } from "next/server";
+
+export async function middleware(request: NextRequest) {
+  const authToken = request.cookies.get("psg_auth_token")?.value;
+  if (!authToken) {
+    return NextResponse.redirect(new URL("/", request.url));
+  }
```
Comment (iamr-kumar) [inline 03-Login-app-directory/middleware.ts:8]: This only checks if `authToken` exists in the cookie. The token could be invalid and a check should be added for that.

Ideally the following should be added:
```
const authToken = request.cookies.get('psg_auth_token')?.value;

  if (!authToken) {
    return NextResponse.redirect(new URL('/auth', request.url));
  }
  const userId = await passage.validAuthToken(authToken);
  if (!userId) {
    return NextResponse.redirect(new URL('/auth', request.url));
  }
```

but for some reason, `passage.validAuthToken` is always returning `undefined` even though a valid token is present.

**Review by himichaelroberts — APPROVED** [review]

**Pull request merged**